### ISSUE TITLE
RMK section of a TAF leading to bogus weather conditions

### ISSUE THREAD

**Issue by aeharding** [mivek/MetarParser#329]
**Describe the bug**

I am noticing some bogus weather conditions being picked up from the RMK section of a TAF.

**To Reproduce**

1. Go to https://www.metar-taf-decoder.com/taf and input the following TAF (simplified to remove other trends, otherwise a valid TAF found in the wild):

```
TAF CYTL 121940Z 1220/1308
  TEMPO 1303/1308 2SM -SN RMK FCST BASED ON AUTO OBS. FCST BASED ON OBS BY OTHER SRCS. WIND SENSOR INOP. NXT FCST BY 130200Z
```

2. Notice funnel cloud weather conditions:

<img width="494" alt="image" src="https://user-images.githubusercontent.com/2166114/163089569-28f56c9e-c570-40dd-b00e-eeea3d53898b.png">

**Expected behavior**

No funnel cloud weather conditions.

**Additional context**

It appears to be picked up from the "remark" section. When I remove everything after `RMK` the problem goes away.

I believe the Python implementation has the same issue.

**Comment by aeharding** [comment]
```diff
diff --git a/src/parser/parser.ts b/src/parser/parser.ts
index 1e548a8..3644985 100644
--- a/src/parser/parser.ts
+++ b/src/parser/parser.ts
@@ -486,8 +486,10 @@ export class TAFParser extends AbstractParser {
    */
   parseTrend(index: number, line: string[], trend: ITAFTrend) {
     for (let i = index; i < line.length; i++) {
-      if (line[i] === this.RMK) parseRemark(trend, line, i, this.locale);
-      else if (this.#validityPattern.test(line[i]))
+      if (line[i] === this.RMK) {
+        parseRemark(trend, line, i, this.locale);
+        break;
+      } else if (this.#validityPattern.test(line[i]))
         trend.validity = parseValidity(line[i]);
       else super.generalParse(trend, line[i]);
     }
```

fwiw, this seems to have fixed it in my typescript port - just the single `break` statement. I'm not sure if there are any ramifications of stopping any other parsing after seeing the RMK token, though.

**Comment by mivek** [comment]
Hello @aeharding 

Thanks for reporting this bug and the analysis. Indeed in the `updateTrend` method, I iterate on the token but in case of the remark, the rest of the tokens are given as parameter to the `parseRMK` method so it is decoded however the loop variable is not reassigned therefore the program iterates again over the token. The token `FCST` (forecast) is recognized as funnel cloud because of FC.

Adding  a `break` is the easiest fix here. I'll do a PR to fix it

**Comment by mivek** [comment]
Hello @aeharding 

The fix is available in version 2.8.1 of the project. I will do the same update in the python version
